packages/codec/lib/contexts/utils.ts

```diff
@@ -252,6 +252,9 @@ interface CborInfo {
   cborSegment: string;
 }
 
+//note: we *don't* bother handling Vyper 0.3.4's idiosyncratic CBOR
+//here, because it's always fixed, so there just isn't any need to;
+//we're OK with just not normalizing that
```

Comment:
I assume this is a comment for future package maintainers who might be looking for Viper 0.3.4 CBOR handling here?

If it's meant to help consumers of the function, I'm afraid that it isn't prescriptive enough for my feeble brain to understand. Specifically it's not clear to me if I should avoid calling this with binaries produced by Vyper 0.3.4, or if I should expect it to error or return null if I do call it with said Vyper metadata.

If it's easy to detect that the binary contains said CBOR (e.g. with a call to `isValidCBOR` as done in `packages/code-utils/src/index.ts`, perhaps this function should throw some `InvalidCBORError` with a message that indicates that Viper 0.3.4 isn't supported here.

Alternatively if we're already just returning `null` in that case, perhaps this should just be converted to typedoc comments and that can be noted in the `@returns` line.

Reply:
I think I lean toward wanting it to throw, as that's a bit more aligned with a "fail fast and fail loudly" mindset than returning `null` would be. Assuming it does return `null`, prior to the addition of your comment it wouldn't have been very clear as to why that was happening to anyone other than someone very deeply familiar with the ins-and-outs of the various compilers and CBOR.

Reply:
All of that said - the comment is a net improvement over the state of things as you found it, so I'm happy for that to merge as-is.

Reply:
> I assume this is a comment for future package maintainers who might be looking for Viper 0.3.4 CBOR handling here?

Yup!

> If it's meant to help consumers of the function, I'm afraid that it isn't prescriptive enough for my feeble brain to understand. Specifically it's not clear to me if I should avoid calling this with binaries produced by Vyper 0.3.4, or if I should expect it to error or return null if I do call it with said Vyper metadata.

Ah, sorry.  TBH I wasn't really expecting there to be other future consumers of this function; you'll notice it's not exported.

To answer your question: It will return `null`.  What this function does is, attempt to find CBOR info on the end of a binary, return information about it if it's present, and return `null` if it's not.  Since Vyper 0.3.4 does its CBOR in a weird way, this function will fail to detect it and return `null`.  Due to it only being used in this one specific context, however, that's OK, and the point of the comment is to explain why.

Throwin an error instead of returning `null` would of course have been another way to do it, but not what I went with.  Note that detecting the Vyper 0.3.4 situation and doing something different in that case is no easier than detecting the Vyper 0.3.4 situation and returning meaningful information, so doing that would not save any work.

Reply:
I think it's fine as-is. As purely a style preference I tend to like to avoid returning `null`, as I feel that throwing errors communicates the design intent a bit better, however I think it's perfectly acceptable practice here, _especially_ since this is a local function.